core/scheduler/service/replay_worker.go

```diff
@@ -141,7 +141,7 @@ func (w ReplayWorker) processPartialReplayedRequest(ctx context.Context, replayR
 		w.l.Info("cleared [%s] [%s] run for replay %s", replayReq.Replay.JobName().String(), toBeReplayedRuns[0].ScheduledAt, replayReq.Replay.ID().String())
 
 		updatedReplayMap[toBeReplayedRuns[0].ScheduledAt] = scheduler.StateReplayed
```

Comment:
do we call clear method one by one for a partial replay? can we clear all of those pending state replays?

Reply:
yes, we need to clear one by one as the partial replay will only be applicable for `sequential` mode.